contrib/docker/start.apache.sh

```diff
@@ -1,15 +1,39 @@
 #!/bin/bash
 
+# We should already be in /var/www, but just to be explicit
+cd /var/www || exit
+
 # Create the storage tree if needed and fix permissions
 cp -r storage.skel/* storage/
 chown -R www-data:www-data storage/ bootstrap/
 
+if [[ ! -e storage/.docker.init ]]; then
+    echo "Fresh installation, initializing database..."
+    chown www-data:www-data .env &&
+        gosu www-data php artisan key:generate &&
+        gosu www-data php artisan migrate:fresh --force &&
+        gosu www-data php artisan passport:keys &&
+        echo "done" >storage/.docker.init
+fi
+
 # Refresh the environment
-php artisan storage:link
-php artisan horizon:publish
-php artisan route:cache
-php artisan view:cache
-php artisan config:cache
-
-# Finally run Apache
-apache2-foreground
+gosu www-data php artisan storage:link
+gosu www-data php artisan horizon:publish
+gosu www-data php artisan config:cache
+# gosu www-data php artisan cache:clear
+gosu www-data php artisan route:cache
+gosu www-data php artisan view:cache
+
+# Check for migrations
+gosu www-data php artisan migrate:status | grep No && migrations=yes || migrations=no
+
+if [ "$migrations" = "yes" ]; then
```

Comment:
This doesn't give control to the admin to decide when to run the db migrations - which ... I see two sides of the argument. Maybe add an environment variable? e.g. ALLOW_AUTO_MIGRATION or something and only run the code here if that's set? That would give admins the ability to manually run the migration (e.g. after they've backed up the database)